src/lib.rs

```diff
@@ -742,43 +746,41 @@ impl Device {
     }
 
     #[cfg(feature = "log")]
-    /// Prints the PCI configuration space in a structured format similar to `lspci -x`.
+    /// Prints the first 64 bytes of PCI configuration space in a structured format similar to `lspci -x`.
     ///
     /// # Example Output
     ///
     /// 00:19.00 Ethernet Vendor:10EC Device:8168 (rev 10)
     /// 00: 1a bc 3d e4 ...
     /// 10: 4f 5a 1b 3c ...
     ///
-    pub fn print_pci_config_space(&self) {
-        let mut buf = ArrayString::<300>::new();
-
-        let _ = write!(
-            buf,
-            "\n{:02X}:{:02X}.{:02X} {:?} Vendor:{:02X} Device:{:02X} (rev {})",
-            self.bdf.bus.val(),
-            self.bdf.dev.val(),
-            self.bdf.func.val(),
-            self.device_info.desc,
-            self.vendor_id,
-            self.device_id,
-            self.revision,
-        );
+    pub fn print_std_config_space(&self) {
+        self.print_config_space_bytes(64);
+    }
 
-        for i in 0..64 {
-            let value = if cfg!(test) {
-                0x0 + i % 0xff
+    #[cfg(feature = "log")]
+    /// Prints full PCI configuration space in a structured format similar to `lspci -x`.
+    /// It will print 256 bytes for standard PCI Device, and it will print 4096 bytes for
+    /// PCIe devices.
+    ///
+    /// # Example Output
+    ///
+    /// 00:19.00 Ethernet Vendor:10EC Device:8168 (rev 10)
+    /// 00: 1a bc 3d e4 ...
+    /// 10: 4f 5a 1b 3c ...
+    ///
+    pub fn print_full_config_space(&self) {
+        let max_config_space = if cfg!(test) {
+            PCIE_CONFIG_SIZE
+        } else {
+            if let Some(_) = self.probe_capability(CapabilityId::PciExpress) {
```

Comment:
nit: `if self.probe_capability(CapabilityId::PciExpress).is_some()`

Reply:
oops, forgot about .is_some(). Thanks!

Reply:
updated it